### PR TITLE
Compose support

### PR DESCRIPTION
**Pull request by davidliu**

### CODE REVIEW

File: livekit-android-sdk/src/main/java/io/livekit/android/room/Room.kt [hunk 1]
```diff
@@ -58,13 +60,24 @@ constructor(
     @Deprecated("Use events instead.")
     var listener: RoomListener? = null
 
-    var sid: Sid? = null
-        private set
-    var name: String? = null
+    /**
+     * Changes can be observed by using [io.livekit.android.util.flow]
+     */
+    var sid: Sid? by flowDelegate(null)
```
Comment (davidliu) [inline livekit-android-sdk/src/main/java/io/livekit/android/room/Room.kt:66]: These flowDelegates are the guts of the compose support, allowing them to be accessed normally like variables, but also observed using the `flow` function to get notified of changes (and can be used in compose for state purposes).

File: sample-app-compose/src/main/java/io/livekit/android/composesample/ParticipantItem.kt [hunk 2]
```diff
@@ -1,80 +1,116 @@
...
-        return participant
-            .videoTracks.values
-            .firstOrNull()?.track as? RemoteVideoTrack
-    }
 
-    fun setupVideoIfNeeded(videoTrack: RemoteVideoTrack, view: TextureViewRenderer) {
-        if (videoBound) {
-            return
-        }
+    val identity by participant::identity.flow.collectAsState()
```
Comment (davidliu) [inline sample-app-compose/src/main/java/io/livekit/android/composesample/ParticipantItem.kt:34]: example of using the flows in compose

Comment (davidzhao) [reply sample-app-compose/src/main/java/io/livekit/android/composesample/ParticipantItem.kt:34]: this looks neat! how does it know when to refresh?

**Review by davidzhao — APPROVED** [review]
looks good!

**Comment by hiroshihorie** [comment]
same with @boks1971 I'm not fluent in Kotlin but will take a look 😂

File: sample-app-compose/src/main/java/io/livekit/android/composesample/ParticipantItem.kt (hunk 2 above)
Comment (davidliu) [reply sample-app-compose/src/main/java/io/livekit/android/composesample/ParticipantItem.kt:34]: So kotlin flows are basically an observable stream, and in this case it emits whenever the value is updated. That gets turned into a state variable with the `collectAsState()` method here.

On the compose side, there's just a huge amount of behind the scenes compiler side magic that goes on, but basically whenever a state variable is changed, it triggers a UI update.

**Comment by davidliu** [comment]
> lgtm @davidliu . I really hope someday I will learn Kotlin/Swift etc. to give more useful review feedback :-) Currently, I am just looking for something that kinda looks incorrect :-)

Haha, always good to have more eyes. Feel free to ask any questions on the kotlin stuff, especially since this one's a little crazy on the language specific stuff.

**Pull request merged**